File: 01/problem_01.py
import sys

groups = []
with open(sys.argv[1], 'rt') as input_file:
    group = []
    for line in input_file:
        line = line.strip()
        if line == "" and group:
            groups.append(group)
            group = []
        else:
            group.append(int(line))
    if group:
        groups.append(group)

totals = [sum(g) for g in groups]
print(max(totals))

print("")
stotals = sorted(totals, reverse=True)
for t in stotals[:3]:
    print(t)

print("")
print(sum(stotals[:3]))

File: 01/problem_01.rs
use std::io;
use std::io::prelude::*;
use std::io::BufReader;
use std::fs::File;


fn main() -> io::Result<()> {
    let f = File::open("input.txt")?;
    let reader = BufReader::new(f);

    let mut totals: Vec<i32> = Vec::new();
    let mut accum: Vec<i32> = Vec::new();

    for line in reader.lines() {
        let linev: String = line?;
        let trimmed: &str = linev.trim();
        if trimmed == "" {
            totals.push(accum.iter().sum());
            accum.clear();
        } else {
            accum.push(trimmed.parse::<i32>().expect("Expected integer"));
        }
    }
    totals.push(accum.iter().sum());

    totals.sort();
    totals.reverse();
    println!("{}", totals[0]);
    println!("{}", totals[0..3].iter().sum::<i32>());
    return Ok(());
}